src/ufuncs.c

```diff
@@ -2425,10 +2430,13 @@ int init_ufuncs(PyObject* m, PyObject* d) {
   DEFINE_Yd_Y(simplify);
   DEFINE_Yd_Y(simplify_preserve_topology);
 
-  DEFINE_YY_Y(intersection);
+  // 'REORDERABLE' sets PyUFunc_ReorderableNone instead of PyUFunc_None as 'identity',
+  // meaning that the order of arguments does not matter. This enables reduction over
+  // multiple (or all) axes.
+  DEFINE_YY_Y_REORDERABLE(intersection);
   DEFINE_YY_Y(difference);
-  DEFINE_YY_Y(symmetric_difference);
-  DEFINE_YY_Y(union);
+  DEFINE_YY_Y_REORDERABLE(symmetric_difference);
+  DEFINE_YY_Y_REORDERABLE(union);
```

Comment:
For union it's not needed strictly for `union_all` to make this a reorderable ufunc, but this enables `pygeos.lib.union.reduce(.., axis=None)` ?

Reply:
It is indeed not implemented like the others. But still, I like to be consistent here, union is reorderable.